action/action.proto

```diff
@@ -7,24 +7,37 @@ option java_outer_classname = "ActionProto";
 
 service ActionService {
     rpc Arm(ArmRequest) returns(ActionResult) {}
-    rpc TakeOff(TakeOffRequest) returns(ActionResult) {}
+    rpc Disarm(DisarmRequest) returns(ActionResult) {}
+    rpc Takeoff(TakeoffRequest) returns(ActionResult) {}
     rpc Land(LandRequest) returns(ActionResult) {}
+    rpc Kill(KillRequest) returns(ActionResult) {}
+    rpc ReturnToLaunch(ReturnToLaunchRequest) returns(ActionResult) {}
+    rpc TransitionToFixedWings(TransitionToFixedWingsRequest) returns(ActionResult) {}
+    rpc TransitionToMulticopter(TransitionToMulticopterRequest) returns(ActionResult) {}
+    rpc GetTakeoffAltitude(GetTakeoffAltitudeRequest) returns(TakeoffAltitudeResult) {}
+    rpc SetTakeoffAltitude(SetTakeoffAltitudeRequest) returns(ActionResult) {}
+    rpc GetMaximumSpeed(GetMaximumSpeedRequest) returns(MaximumSpeedResult) {}
+    rpc SetMaximumSpeed(SetMaximumSpeedRequest) returns(ActionResult) {}
 }
 
-message UUID {
-    uint64 value = 1;
-}
+message ArmRequest {}
+message DisarmRequest {}
+message TakeoffRequest {}
+message LandRequest {}
+message KillRequest {}
+message ReturnToLaunchRequest {}
+message TransitionToFixedWingsRequest {}
+message TransitionToMulticopterRequest {}
+message GetTakeoffAltitudeRequest {}
```

Comment:
Just For Me, is it necessary to have individual message for each Action request though they're empty ? I mean can't we have [Empty message](https://github.com/google/protobuf/blob/master/src/google/protobuf/empty.proto) ?

Reply:
I am new to protobuf, but I saw this somewhere as a "good practice". The idea is that if you define your protocol saying that it takes an empty message, you will **never** be able to add anything to the request.

If you define a specific message for the request, you could later add parameters to it without breaking the API. And I don't think it hurts performances, as it just generates a few more "empty" messages.

One quick example would be, say, if we wanted to add the "UUID" in all the messages, later. Or something specific to "TransitionToFixedWings".

Reply:
Ok, seems good to me.

Reply:
Actually, this advice came from Kenton Varda, [here on StackOverflow](https://stackoverflow.com/questions/31768665/can-i-define-a-grpc-call-with-a-null-request-or-response):

> ... we as developers are really bad at guessing what we might want in the future. So I recommend being safe by always defining custom params and results types for every method, even if they are empty.

I think I will also add custom responses...

Reply:
OK.